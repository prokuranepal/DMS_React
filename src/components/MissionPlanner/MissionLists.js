import React from 'react';
import Grid from '@material-ui/core/Grid';
import { makeStyles } from '@material-ui/core/styles';
import {Typography } from '@material-ui/core';
import MissionList from './MissionList/MissionList';
const useStyles = makeStyles((theme) => ({
    root: {
        width: '100%',
        backgroundColor:'#A5D8DD',
        // height: '93vh'
    },
    subRoot: {
        width: '80%',
        margin: '20px auto',
        backgroundColor: '#A5D8DD',
        // minHeight: '50vh'
    },
    header: {
        minHeight: '50px',
        margin: 'auto'
    },
    text: {
        padding: theme.spacing(3),
        
    }
}));

const missions = [
    {
        id: 'abc',
        source: 'Dharan',
        destination: 'Biratnagar'
    },
    {
        id: 'xyz',
        source: 'Dharan',
        destination: 'Biratnagar'
    },
    {
        id: 'efg',
        source: 'Dharan',
        destination: 'Biratnagar'
    },
    {
        id: 'mno',
        source: 'Dharan',
        destination: 'Biratnagar'
    },
    {
        id: 'mnop',
        source: 'Dharan',
        destination: 'Biratnagar'
    },
    {
        id: 'mnopq',
        source: 'Dharan',
        destination: 'Biratnagar'
    }
]
const MissionLists = props => {
    const classes = useStyles();
    return(
    <Grid container className={classes.root}>
        <Grid container className={classes.subRoot}>
            <div className={classes.header}>
            <Typography variant="h4" className={classes.text}>MISSION LISTS</Typography>
            </div>
                {missions.map(mission => {
                    return <MissionList key={mission.id} id={mission.id} source={mission.source} destination={mission.destination}/>
                })}
        </Grid>
    </Grid>)
}

export default MissionLists;