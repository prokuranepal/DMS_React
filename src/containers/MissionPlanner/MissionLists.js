import React, { useEffect } from "react";
import MaterialTable from 'material-table';
import TableIcons from '../../homeComponents/TableIcons/TableIcons';
import * as actions from '../../store/actions/mission';
import {useDispatch, useSelector} from 'react-redux';
import { Redirect } from 'react-router';

const CreateData = (data) => {
    return {_id:data._id, name:data.name, estimated_time: data.estimated_time, distance: data.distance, destination: data.destination.name, location: data.destination.location,
    waypoints: data.wb, origin: data.hospital.name }
}

/**
* A List of missions
*
* @param - No parameters
* @returns {MissionList} - returns a material table of missions
*/

const MissionList = () => {

    const [state, setState] = React.useState({
        columns: [
            { title: 'Name', field: 'name' },
            { title: 'Estimated Time', field: 'estimated_time'},
            { title: 'Distance', field: 'distance'},
            { title: 'Destination', field: 'destination' },
            { title: 'Location', field: 'location' },
            { title: 'Waypoints', field: 'waypoints' },
            { title: 'Origin', field: 'origin' },
        ],
        data: [
        ],
    });
    const [selectedRow, setSelectedRow] = React.useState(null);
    const [redirect, setRedirect] = React.useState(null);
    const missions = useSelector(({mission}) => mission.missions);
    const dispatch = useDispatch();

    useEffect(() => {
        const refinedMissions = missions.map(mission => {
            return CreateData(mission);
        })
        console.log(missions);
        setState(prevState => { return { ...prevState, data: refinedMissions } })
    }, [missions])

    const setSelected = (missionId) => {
        console.log(missionId)
        dispatch(actions.getMission(missionId));
        setRedirect(<Redirect
            to={{
            pathname: "/app/missionplanner/missionview",
            state: { edit: true }
          }}
        />)
    }

    useEffect(() => {
        dispatch(actions.fetchMissionList());
    },[dispatch])

    return (
        <div className="app-wrapper">
            <div className="animated slideInUpTiny animation-duration-3">
            {redirect}
                <MaterialTable
                    title="Mission List"
                    columns={state.columns}
                    data={state.data}
                    data-test="missiontable"
                    icons={TableIcons}
                    onRowClick={((evt, selectedRow) => setSelected(selectedRow._id))}
                    options={{
                        rowStyle: rowData => ({
                            backgroundColor: (selectedRow === rowData.tableData.id) ? '#EEE' : '#FFF',
                            font: 'Roboto'
                        }),
                        headerStyle: {
                            backgroundColor: '#01579b',
                            color: '#FFF'
                          }
                    }}
                    editable={{
                        onRowDelete: (oldData) =>
                            new Promise((resolve) => {
                                setTimeout(() => {
                                    resolve();
                                    setState((prevState) => {
                                        const data = [...prevState.data];
                                        data.splice(data.indexOf(oldData), 1);
                                        dispatch(actions.deleteMission())
                                        return { ...prevState, data };
                                    });
                                }, 600);
                            }),
                    }}
                />
            </div>
        </div>
    );
}

export default MissionList;