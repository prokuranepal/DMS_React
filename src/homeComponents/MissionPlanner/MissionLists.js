import React from 'react';
import Grid from '@material-ui/core/Grid';
import { makeStyles } from '@material-ui/core/styles';
import {Typography } from '@material-ui/core';
import MissionList from './MissionList/MissionList';
const useStyles = makeStyles((theme) => ({
    root: {
        width: '100%',
        backgroundColor:'#A5D8DD',
        // height: '93vh'
    },
    subRoot: {
        width: '80%',
        margin: '20px auto',
        backgroundColor: '#A5D8DD',
        // minHeight: '50vh'
    },
    header: {
        minHeight: '50px',
        margin: 'auto'
    },
    text: {
        padding: theme.spacing(3),
        
    }
}));

export const missions = [
    {
        id: 'abc',
        source: 'Dharan',
        destination: 'Biratnagar'
    },
    {
        id: 'xyz',
        source: 'Dharan',
        destination: 'Biratnagar'
    },
    {
        id: 'efg',
        source: 'Dharan',
        destination: 'Biratnagar'
    },
    {
        id: 'mno',
        source: 'Dharan',
        destination: 'Biratnagar'
    },
    {
        id: 'mnop',
        source: 'Dharan',
        destination: 'Biratnagar'
    },
    {
        id: 'mnopq',
        source: 'Dharan',
        destination: 'Biratnagar'
    }
]

/**
* A List of missions
*
* @param - No parameters
* @returns {MissionList} - returns a material table of missions
*/
const MissionLists = props => {
    const classes = useStyles();
    return(
    <Grid container className={classes.root}>
        <Grid container className={classes.subRoot}>
            <div className={classes.header}>
            <Typography variant="h4" className={classes.text}>MISSION LISTS</Typography>
            </div>
                {missions.map(mission => {
                    return <MissionList key={mission.id} id={mission.id} source={mission.source} destination={mission.destination} data-test="mission-component"/>
                })}
        </Grid>
    </Grid>)
}

export default MissionLists;