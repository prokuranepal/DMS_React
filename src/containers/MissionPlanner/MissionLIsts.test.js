import React from 'react';
import MissionLists,{missions} from './MissionLists';
import {
    configure,
    mount
} from 'enzyme';
import EnzymeAdapter from 'enzyme-adapter-react-16';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
const mockStore = configureStore([]);
// JestHook.mock('expo-font');
configure({
    adapter: new EnzymeAdapter
})


const setup = (props = {},store=null, state = null) => {
    return mount(<Provider store={store} > < MissionLists {...props}  /></Provider>)
    
}

const findByTestAttr=(wrapper, val)=>{
    return wrapper.find(`[data-test='${val}']`)
}



describe('MissionLists />', () => {
  let wrapper;
let store

  beforeEach(() => {

store = mockStore({
  mission:{
missions:[{"_id":1,"data":"data1", name:"name1", estimated_time:"20:45", distance:10, destination:{name:"destination1", location:"location1"}, wb:"wb1", hospital:{name:"hospital1"}},
{"_id":2,"data":"data2",name:"name2", estimate_time:"20:42", distance:12, destination:{name:"destination2", location:"location2"}, wb:"wb2", hospital:{name:"hospital2"}}]
}}
);
store.dispatch=jest.fn();

    wrapper = setup({},store)
  });

  afterEach(() => {
    jest.clearAllMocks();
  });


it("MissionLists snapshot", () => {
        // let wrapper = setup(dummy_data);
        expect(wrapper).toMatchSnapshot();

  
})

it("MissionLists Components", () => {

   const missionComponent = findByTestAttr(wrapper, "missiontable")
    // expect(missionComponent).toHaveLength(missions.length);
    // expect(missionComponent.at(2).prop("id")).toEqual("efg");
    // expect(missionComponent.at(3).prop("source")).toEqual("Dharan");
    // expect(missionComponent.at(0).prop("destination")).toEqual("Biratnagar");
    expect(missionComponent).toHaveLength(3)
    expect(missionComponent.at(2).prop('data')).toEqual(
      [{"destination": "destination1", "distance": 10, "estimated_time": "20:45", "location": "location1", "name": "name1", "origin": "hospital1", "tableData": {"id": 0}, "waypoints": "wb1"}, {"destination": "destination2", "distance": 12, "estimated_time": undefined, "location": "location2", "name": "name2", "origin": "hospital2", "tableData": {"id": 1}, "waypoints": "wb2"}]
  )


})
})
