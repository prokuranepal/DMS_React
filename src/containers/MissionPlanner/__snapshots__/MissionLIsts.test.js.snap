// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MissionLists /> MissionLists snapshot 1`] = `ReactWrapper {}`;
